### ISSUE TITLE
fread's nrow argument could accept -ve values to skip last 'n' rows

### ISSUE THREAD

**Issue by arunsrinivasan** [Rdatatable/data.table#1643]
could be useful for this post: http://stackoverflow.com/q/36558437/559784

```r
fread(file, nrow=-3) # could skip last 3 lines, for example.
```

**Comment by MichaelChirico** [comment]
Note that this would break the current default of `-1L`.

**Comment by franknarf1** [comment]
Yeah, I could use this. I'm currently reading in files that often have an incomplete last line (not enough fields as inferred from commas), which reliably causes fread to crash R. 

It would be nice to set `skip.last=1L` to avoid this. Because `nrow` already allows a negative value as Michael mentioned, I think it would be cleaner as a separate arg or allowing the `skip` arg to have a length of two (with the second component of the vector taking on this role when present).

**Comment by jangorecki** [comment]
I would prefer the way mentioned by Arun, as it would be consistent to linux `head` and `tail` way of handling negative values. If negative `skip` is currently being, and cannot be easily changed, then it make sense to allow `skip` of length two makes, so `skip=c(0, 1)` would skip just the last line.
Current workaround, just for completeness: `fread("head -n -1 filename")`.

**Comment by lanceculnane** [comment]
On a related note, it would be nice if we could pass a list of indicies (also as part of the 'skip' parameter) to explicitly read in the rows you want, like we can do in python's Pandas. If the list of indicies is random, it is a nice way to create a random sample of a data frame which is too large to be read onto a local machine, for instance.

**Comment by MichaelChirico** [comment]
@lanceculnane see also #583
